packages/overlays/src/LocalOverlayController.js

```diff
@@ -128,10 +126,15 @@ export class LocalOverlayController {
     this.isShown ? this.hide() : this.show();
   }
 
-  _createOrUpdateOverlay(shown = this._prevShown, data = this._prevData) {
+  async _createOrUpdateOverlay(shown = this._prevShown, data = this._prevData) {
     if (shown) {
       this._contentData = { ...this._contentData, ...data };
 
+      // preload popper before render, so the user does not see a different view between render and popper being applied
+      if (!this.popperModule) {
+        await this.__preloadPopper();
+      }
```

Comment:
my expectation was that the async nature of positioning does not impact the sync nature of the interface

or to be more precisely: we can tolerate the fact that the overlay still has wrong position while all the other things behave correctly, like you can pass and render data, A11Y attributes are all set, etc...

now it feels like that this is not correct always, e.g. line 370 in this test should not be needed

https://github.com/ing-bank/lion/blob/04af7fdd7506c2a6a9af3efb6876179bc0b0d3cd/packages/overlays/test/LocalOverlayController.test.js#L366-L373

in general I would expect only the tests that check the position to wait, and all other to continue to work synchronously

Reply:
Ah it makes more sense now reading this. You're absolutely right I think, just because it hasn't positioned correctly yet, doesn't mean we cannot already test non-positioning related stuff. I have removed unnecessary awaits from the test, let me know if it's more like you would expect now